src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -146,6 +132,49 @@ typedef enum {
     UCT_IB_MLX5_MMIO_MODE_LAST
 } uct_ib_mlx5_mmio_mode_t;
 
+typedef enum {
+    UCT_IB_MLX5_MMIO_TYPE_VERBS,
+    UCT_IB_MLX5_MMIO_TYPE_DEVX
+} uct_ib_mlx5_mmio_type_t;
+
+/* Blue flame register */
+typedef struct uct_ib_mlx5_mmio_reg {
+    union {
+        void                    *ptr;
+        uintptr_t               uint;
+    } addr;
+    uint8_t                     mode;
+    uint8_t                     type;
+} uct_ib_mlx5_mmio_reg_t;
+
+typedef struct uct_ib_mlx5_mmio_reg_verbs {
+    uct_worker_tl_data_t        super;
+    uct_ib_mlx5_mmio_reg_t      data;
+} uct_ib_mlx5_mmio_reg_verbs_t;
+
+typedef struct uct_ib_mlx5_mmio_reg_devx {
+#if HAVE_DEVX
+    struct mlx5dv_devx_uar      *uar;
+    uct_ib_mlx5_mmio_reg_t      data;
+#elif GTEST_UCX
```

Comment:
need to avoid this

Reply:
Don't want gtest only things become part of release code .

Reply:
always inherit from uct_worker_tl_data_t   , no need dummy for C++